spec/controllers/post_controller_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+require 'rails-controller-testing'
+Rails::Controller::Testing.install
+
+RSpec.describe PostsController, type: :request do
+  describe 'GET #index' do
+    it 'returns a successful response' do
+      get '/users/:user_id/posts'
+      expect(response).to be_successful
+      expect(response).to have_http_status(200)
+    end
+
+    it 'renders the correct template' do
+      get '/users/:user_id/posts'
+      expect(response).to render_template(:index)
+    end
+
+    it 'includes the correct placeholder text' do
+      get '/users/:user_id/posts'
+      expect(response.body).to include('List of posts')
+    end
+  end
```

Comment:
- [ ] As I mentioned in the above comment, kindly ensure that you add tests to both `index` and `show` actions for `User` and `Post` controllers. Right now I can see that you only test the `index` action, I suggest you also add tests to the `show` action with the following checks as you did for the `index`:
  * If response status was correct.
  * If a correct template was rendered.
  * If the response body includes correct placeholder text.
```ruby
# Example

describe 'GET /index' do
  # Tests for the index action
end

describe 'GET /show' do
  # Tests for the show action
end
```